dbms/src/IO/ReadHelpers.cpp

```diff
@@ -1028,4 +1028,18 @@ void skipToUnescapedNextLineOrEOF(ReadBuffer & buf)
     }
 }
 
+bool safeInBuffer(ReadBuffer & in, DB::Memory<> & memory, char * & begin_pos, bool force)
```

Comment:
What does this do, and what "force" means? Needs a comment and probably a better name.